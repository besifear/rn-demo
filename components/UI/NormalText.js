import React from 'react';
import { StyleSheet, View, Text } from 'react-native';

const NormalText = props => {
  return (
    <View>
      <Text style={styles.normal}>{props.children}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  normal: {
    paddingHorizontal: 10,
    fontFamily: 'open-sans',
    fontSize: 14
  }
})

export default NormalText;
