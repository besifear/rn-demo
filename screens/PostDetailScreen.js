import React from 'react';
import { StyleSheet, View, Text } from 'react-native';

const PostDetailScreen = props => {
  return (
    <View style={styles.screen}>
      <Text>POST DETAIL SCREEN!!!</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  screen: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  }
})

export default PostDetailScreen;
