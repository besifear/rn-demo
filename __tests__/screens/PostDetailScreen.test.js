import React from 'react';
import { Text, Button } from 'react-native';
import renderer from 'react-test-renderer';

import PostDetailScreen from '../../screens/PostDetailScreen';
import NormalText from '../../components/UI/NormalText';

jest.mock('react-redux', () => ({
  useDispatch: jest.fn().mockImplementation(() => () => {}),
  useSelector: jest.fn(),
}));

import { useSelector } from 'react-redux';

const mockedProps = {
  navigation: {
    getParam: () => {},
    navigate: () => {},
  }
};

describe('<PostDetailScreen />', () => {
  let tree;
  beforeEach(() => {
    tree = renderer.create(<PostDetailScreen {...mockedProps} />);
  });

  it('has 1 child', () => expect(tree.toJSON().children.length).toBe(1));
  it('renders correctly', () => expect(tree.toJSON()).toMatchSnapshot());
  it('renders body of post', () => {
    useSelector.mockImplementation(() => ({ id: 1, title: 'Post 1', body: 'test-body-1' }));
    tree = renderer.create(<PostDetailScreen {...mockedProps} />);
    expect(tree.root.findAllByType(NormalText).length).toBe(1);
    expect(tree.root.findByType(NormalText).props.children).toBe('test-body-1')
  });
  it('has a delete button', () => {
    useSelector.mockImplementation(() => ({ id: 1, title: 'Post 1', body: 'test-body-1' }));
    tree = renderer.create(<PostDetailScreen {...mockedProps} />);
    expect(tree.root.findAllByType(Button).length).toBe(1);
    expect(tree.root.findByType(Button).props.title).toBe("Delete Post");
  });
  it('renders post not found text', () => {
    useSelector.mockImplementation(() => (null));
    tree = renderer.create(<PostDetailScreen {...mockedProps} />);
    expect(tree.root.findByType(Text).props.children).toBe("Post not found!");
  });
})
