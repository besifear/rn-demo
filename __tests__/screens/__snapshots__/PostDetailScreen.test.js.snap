// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostDetailScreen /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Post not found!
  </Text>
</View>
`;
